### ISSUE TITLE
 reproducing issue

### ISSUE THREAD

**Issue by jueliangguke** [hantek/SelfAttentiveSentEmbed#3]
Hi, 
I try to implement Self-Attentive model  with pytorch and test it on SNLI dataset but the accuracy is up to about 80.2% (I have revised some details according to your codes which are different from those in paper). Then I run your code with the command you give, 

> python segae_l2_dpout.py 300 150 3000 30 0.01 0.0001 0.3 1. 300 50 0.5 5 0.1 1

the accuracy is only about 80.4%. I found the training was underfitting and changed the max epoch number to 15 from 5, then the accuracy improved to about 81.2% (I also tried more training epoches and 
no improvement was obtained ), but is still much worse than 84.4% which was reported in paper.

So how can I reproduce the model on SNLI? Can you give me some advise?

**Comment by hantek** [comment]
Hi jueliangguke,

Thanks for reporting this! I have found that the 84.4% result is generated by an earlier version of the `segae_l2_dpout.py` file. I have thus put that earlier version into the repo, named as `segae_gaereg.py`. The main difference is the place where the network gets regularized. 

If you run the new file with the command 
> python segae_gaereg.py 300 150 4000 30 0.01 0.1 0.5 300 50 100 12 0.1

you should get exactly the same results (the random seed difference between each run may cause some minor fluctuations of course).

I am also attaching the original log file, named as `b8d1e8fd0ec79f9c9159dad0c23d642ae2d5ce69076f2972b2af19e9c25b6197.out`, which generates the 84.4% results reported in the paper. Hope that provides the details you might need in reproduction!

Best,
Zhouhan

**Comment by jueliangguke** [comment]
Hi Zhouhan:

Thanks for your answer! I have run the new file with the new command and get the 84.4% results reported in the paper! 

However, with the same hyperparameter and regularization, my Pytorch code can only get about 81.2%. I am stilling trying...... 

Best, 
jueliangguke

**Comment by hantek** [comment]
Closing because the reproducing issue has been resolved, and no further update shows up.
